### PR TITLE
Softmax added to tensor, torch and docs 

### LINKED ISSUE #91107: Tensor.softmax() wasn't include in documentation
### 📚 The doc issue

Hi pytorch developers!

I've just noticed that Tensors have softmax() method, but it wasn't include in documentation. Is there any reason why it is missing? 

E.g. ``` torch.rand(2, 2).softmax(-1) ```

### Suggest a potential alternative/fix

_No response_

cc @svekars @carljparker

### PR DESCRIPTION
**Pull request by ShisuiUzumaki**
Fixes #91107 

Added `softmax` docs in 

- `pytorch/torch/_tensor_docs.py`
- `pytorch/torch/_torch_docs.py `
- `pytorch/docs/XXX.rst` files. Here XXX represents all those files where I made the change

Although I have added `softmax` in `docs` directory, I was not sure which files/folders required the edits so there could be issues

cc: @lezcano @albanD 

cc @svekars @carljparker @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen

### CODE REVIEW

File: docs/source/tensors.rst [hunk 1]
```diff
@@ -665,6 +665,8 @@ Tensor class reference
     Tensor.atanh_
     Tensor.arctanh
     Tensor.arctanh_
+    Tensor.softmax
+    Tensor.softmax_
```
Comment (lezcano) [inline docs/source/tensors.rst:669]: Does this function exist?

File: torch/_tensor_docs.py [hunk 2]
```diff
@@ -5285,6 +5285,15 @@ def callable(a, b) -> number
 """,
 )
 
+add_docstr_all(
+    "softmax",
+    r"""
+softmax(dim) -> Tensor
+
+See :func:`torch.softmax`
```
Comment (lezcano) [inline torch/_tensor_docs.py:5293]: point to `torch.nn.funcional.softmax`.

File: docs/source/torch.rst [hunk 3]
```diff
@@ -385,6 +385,7 @@ Pointwise Ops
     subtract
     tan
     tanh
+    softmax
```
Comment (lezcano) [inline docs/source/torch.rst:388]: In alphabetical order please.

File: docs/source/tensors.rst [hunk 4]
```diff
@@ -665,6 +665,8 @@ Tensor class reference
     Tensor.atanh_
     Tensor.arctanh
     Tensor.arctanh_
+    Tensor.softmax
```
Comment (lezcano) [inline docs/source/tensors.rst:668]: alphabetical order unless there's a good reason.

File: docs/source/name_inference.rst [hunk 5]
```diff
@@ -222,6 +222,7 @@ If you don't see an operation listed here, but it would help your use case, plea
    :meth:`Tensor.tan_`,None
    ":meth:`Tensor.tanh`, :func:`torch.tanh`",:ref:`keeps_input_names-doc`
    :meth:`Tensor.tanh_`,None
+   ":meth:`Tensor.softmax`, :func:`torch.tanh`",:ref:`keeps_input_names-doc`
```
Comment (lezcano) [inline docs/source/name_inference.rst:225]: alphabetical order.

Comment (lezcano) [reply docs/source/name_inference.rst:225]: also, please be careful when copy pasting.
```suggestion
   ":meth:`Tensor.softmax`, :func:`torch.softmax`",:ref:`keeps_input_names-doc`
```

File: docs/source/tensors.rst (hunk 1 above)
Comment (ShisuiUzumaki) [reply docs/source/tensors.rst:669]: > Does this function exist?

are you referring to `Tensor.softmax_`

**Comment by ShisuiUzumaki** [comment]
@lezcano Please review if I have made the required changes correctly

**Comment by lezcano** [comment]
the commit is polluted

**Review by lezcano — APPROVED** [review]
Cool! Now you just have to sign the CLA and we should be good to go.
If you signed it already, I think that your commits may have got polluted, so squashing them / reseting the author may help.

**Comment by ShisuiUzumaki** [comment]
> Cool! Now you just have to sign the CLA and we should be good to go. If you signed it already, I think that your commits may have got polluted, so squashing them / reseting the author may help.

I am having some trouble with this. I have signed the CLA so many times and tried changing author using `git rebase`, but it does seem to work

**Comment by lezcano** [comment]
Consider squashing the PRs and then doing `git commit --ame --reset-author`

**Comment by ShisuiUzumaki** [comment]
@lezcano For some reason, now CircleCI build is failing. Do you have any idea?

**Comment by ShisuiUzumaki** [comment]
> /easycla

I think all the checks are clear now @lezcano

**Comment by ShisuiUzumaki** [comment]
> @pytorchbot merge -f "just docs"

Thanks for all the help @lezcano it was a great learning experience 😊

**Comment by lezcano** [comment]
wow, that's unexpected. Do you have more details @weiwangmeta?

**Comment by ShisuiUzumaki** [comment]
> wow, that's unexpected. Do you have more details @weiwangmeta?

Yes, if there are any changes needed, please let me know

**Comment by weiwangmeta** [comment]
@ShisuiUzumaki  @lezcano Looks like the internal distributed test may have some non-deterministic failure. Let's plan on getting this in again.
